classes/submission/Collector.inc.php

```diff
@@ -88,17 +89,29 @@ public function filterByCategoryIds(?array $categoryIds): AppCollector
      * Limit results to submissions that contain any pub objects (e.g. publication and galley) with these statuses
      *
      * @param array|null $statuses One or more of DOI::STATUS_* constants
-     * @param bool $strict Whether null/empty values should count when considering Doi::STATUS_UNREGISTERED
      *
      */
-    public function filterByDoiStatuses(?array $statuses, bool $strict = false): AppCollector
+    public function filterByDoiStatuses(?array $statuses): AppCollector
     {
         $this->doiStatuses = $statuses;
-        $this->strictDoiStatusFilter = $strict;
 
         return $this;
     }
 
+    /**
+     * Limit results to submissions that do/don't have any DOIs assign to their sub objects
+     *
+     * @param bool|null $hasDois
+     * @param array $enabledDoiTypes TYPE_* constants to consider when checking submission has DOIs
+     * @return AppCollector
+     */
+    public function filterByHasDois(?bool $hasDois, array $enabledDoiTypes = [\APP\doi\Repository::TYPE_PUBLICATION]): AppCollector
```

Comment:
`[\APP\doi\Repository::TYPE_PUBLICATION]` is great. However, repo facade allows anyone to swap out one repo class for another. To be really safe here, we should always go through the facade. 

I'd recommend leaving `$enabledDoiTypes` null if not set, and then setting the default in the method by going through the `Repo::doi()` facade. If someone wants to override the default to perform a "reset", they can pass an empty array.